trust-store-trust-policy-specification.md

```diff
@@ -1,154 +1,143 @@
-
-- `fail-open`: If revocation endpoint is not reachable, consider artifact as not revoked.
-- `fail-close`: If revocation endpoint is not reachable, consider artifact as revoked.
-
-### Trust Policy Schema
-
-The trust policy is represented as JSON data structure as shown below:
+Trust policy for the scenario where ACME Rockets uses artifacts signed by themselves, and signed and unsigned artifacts from Wabbit Networks:
 
 ```jsonc
 {
     "version": "1.0",
     "trustPolicies": [
         {
-            "name": "verify-signature",
-            "scopes": [
-                "registry.wabbit-networks.io/software/net-monitor"
-                "registry.wabbit-networks.io/software/net-logger" ],
-            "trustStores": [ "trust-store-name-1", "trust-store-name-2" ],
-            "trustAnchors": [
-                "subject: C=US, ST=WA, L=Seattle, O=acme-rockets.io"
-            ],
-            "expiryValidations": {
-                "signatureExpiry": "enforce | warn",
-                "signingIdentityExpiry": "enforce | warn",
-                "timestampExpiry": "enforce | warn"
-            },
-            "revocationValidations": {
-                "signingIdentityRevocation": "enforceWithFailOpen | enforceWithFailClose | warn | skip",
-                "timestampRevocation": "enforceWithFailOpen | enforceWithFailClose | warn | skip"
+            # Policy for set of artifacts signed by Wabbit Networks
+            # that are pulled from ACME Rockets repository
+            "name": "wabbit-networks-images",
+            "registryScopes": [
+                "registry.acme-rockets.io/software/net-monitor"
+                "registry.acme-rockets.io/software/net-logger" ],
+            "signatureVerification" : "strict",
+            "trustedIdentities": {
+                "trustStore": "ca:acme-rockets",
+                "identities": [ 
+                  "x509.subject: C=US, ST=WA, L=Seattle, O=wabbit-networks.io, OU=Security Tools"
+                ]
             }
         },
         {
-            "name": "skip-signature-verification",
-            "scopes": [ "wabbit-networks.io/software/unsigned/productA" ],
-            "skipSignatureVerification": true,
+            # Exception policy for a single unsigned artifact pulled from
+            # Wabbit Networks repository
+            "name": "unsigned-image",
+            "registryScopes": [ "registry.wabbit-networks.io/software/unsigned/net-utils" ],
+            "signatureVerification": "skip",
         },
         {
-            "name": "global-trust-policy",
-            "scopes": [ "*" ],
-            "trustStores": [ "trust-store-name-1", "trust-store-name-2" ],
-            "expiryValidations": {
-                "signatureExpiry": "enforce | warn",
-                "signingIdentityExpiry": "enforce | warn",
-                "timestampExpiry": "enforce | warn"
+            # Policy for with custom verification policy
+            "name": "use-expired-image",
```

Comment:
Should we require * to be added to support all?
Or, should we assume a default scenario where a an entire registry is trusted, unless scoped?

I realize that someone may want to limit docker hub content to library, however most secured environments use private registries, where the registry is the trust boundary, not necessarily a sub-namespace.

Reply:
The `registryScopes` with value '*' makes it explicit to user that any registry is trusted, it's better to be explicit in security related policies, unlike other policies where we may favor conciseness.  We used the same for `identities` where it's now a required property but you can specify a wildcard `*`.

```
"name": "myPolicy",
            "registryScopes": [ "*" ],
            "signatureVerification" : "strict",
            "trustedIdentities": {
                "trustStore": "ca:acme-rockets",
                "identities": [ "*"]
            }
```